es/index.mjs

```diff
@@ -1,270 +0,0 @@
- * @return {number} - delay in milliseconds
- */
-export function exponentialDelay(retryNumber = 0, error, delayFactor = 100) {
-  const delay = Math.pow(2, retryNumber) * delayFactor;
-  const randomSum = delay * 0.2 * Math.random(); // 0-20% of the delay
-  return delay + randomSum;
-}
-
-/** @type {IAxiosRetryConfig} */
-export const DEFAULT_OPTIONS = {
-  retries: 3,
-  retryCondition: isNetworkOrIdempotentRequestError,
-  retryDelay: noDelay,
-  shouldResetTimeout: false,
-  onRetry: () => {}
-};
-
-/**
- * Returns the axios-retry options for the current request
- * @param  {AxiosRequestConfig} config
- * @param  {IAxiosRetryConfig} defaultOptions
- * @return {IAxiosRetryConfigExtended}
- */
-function getRequestOptions(config, defaultOptions) {
-  return { ...DEFAULT_OPTIONS, ...defaultOptions, ...config[namespace] };
-}
-
-/**
- * Initializes and returns the retry state for the given request/config
- * @param  {AxiosRequestConfig} config
- * @param  {IAxiosRetryConfig} defaultOptions
- * @return {IAxiosRetryConfigExtended}
- */
-function getCurrentState(config, defaultOptions) {
-  const currentState = getRequestOptions(config, defaultOptions);
-  currentState.retryCount = currentState.retryCount || 0;
-  config[namespace] = currentState;
-  return currentState;
-}
-
-/**
- * @param  {Axios} axios
- * @param  {AxiosRequestConfig} config
- */
-function fixConfig(axios, config) {
-  if (axios.defaults.agent === config.agent) {
-    delete config.agent;
-  }
-  if (axios.defaults.httpAgent === config.httpAgent) {
-    delete config.httpAgent;
-  }
-  if (axios.defaults.httpsAgent === config.httpsAgent) {
-    delete config.httpsAgent;
-  }
-}
-
-/**
- * Checks retryCondition if request can be retried. Handles it's returning value or Promise.
- * @param  {IAxiosRetryConfigExtended} currentState
- * @param  {Error} error
- * @return {Promise<boolean>}
- */
-async function shouldRetry(currentState, error) {
-  const { retries, retryCondition } = currentState;
-  const shouldRetryOrPromise = currentState.retryCount < retries && retryCondition(error);
-
-  // This could be a promise
-  if (typeof shouldRetryOrPromise === 'object') {
-    try {
-      const shouldRetryPromiseResult = await shouldRetryOrPromise;
-      // keep return true unless shouldRetryPromiseResult return false for compatibility
-      return shouldRetryPromiseResult !== false;
-    } catch (_err) {
-      return false;
-    }
-  }
-  return shouldRetryOrPromise;
-}
-
-/**
- * Adds response interceptors to an axios instance to retry requests failed due to network issues
- *
- * @example
- *
- * import axios from 'axios';
- *
- * axiosRetry(axios, { retries: 3 });
- *
- * axios.get('http://example.com/test') // The first request fails and the second returns 'ok'
- *   .then(result => {
- *     result.data; // 'ok'
- *   });
- *
- * // Exponential back-off retry delay between requests
- * axiosRetry(axios, { retryDelay : axiosRetry.exponentialDelay});
- *
```

Comment:
Is it not useful to keep this in the new code also?

It will still show up in editors when trying to use the function. It's just that we can remove the `{...}` types in `@type`, `@param`, etc.

Reply:
I thought about keeping them, but IMO this should be part of the readme, reading the examples in an editor pop-up is not the best thing.